contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2905,6 +2905,7 @@ handleBatchLevelStatement(TSqlParser::Batch_level_statementContext *ctx, tsqlSel
 	result->body = list_make1(init);
 	// create PLtsql_stmt_execsql to wrap all query string
 	PLtsql_stmt_execsql *execsql = (PLtsql_stmt_execsql *) makeSQL(ctx);
+	execsql->original_query = (makeTsqlExpr(ctx, false))->query;
```

Comment:
It seems that we are passing `char *` from here to `pltsql_store_view_definition` hook. Could you double check the life cycle of this pointer? In other words, `(makeTsqlExpr(ctx, false))->query` is in which memory context?

Reply:
Sorry I did not understand what's your question here. 
At this point, _ctx_ is the batch_level_statement context and it contains the original statement throughout. I am making use of that _ctx_ to get the query and save it in _execsql->original_query_